### PR TITLE
PlanExecution: fix leftover state after preempt

### PR DESCRIPTION
**Pull request by v4hn**
Clearly differentiate between preemption signal and state.

Alternative to #2403.

Fixes https://github.com/ros-planning/moveit_task_constructor/issues/217

### CODE REVIEW

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 1]
```diff
@@ -161,7 +160,8 @@ void plan_execution::PlanExecution::planAndExecute(ExecutableMotionPlan& plan,
 void plan_execution::PlanExecution::planAndExecuteHelper(ExecutableMotionPlan& plan, const Options& opt)
 {
   // perform initial configuration steps & various checks
-  preempt_requested_ = false;
+  bool preempt_requested = false;
+  preempt_.check_and_clear();  // clear any previous preempt_ request
```
Comment (rhaschke) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp:164]: Previously, `preempt_requested_` was reset here. We should keep this behaviour, shouldn't we?

**Review by rhaschke — APPROVED** [review]
This is much cleaner than my solution. I have pushed some suggestions. Please feel free to discard particularly the last one if you don't like the proposed single-line format.

**Review by rhaschke — APPROVED** [review]
You are right: we still need to reset the flag when entering `executeAndMonitor()` directly.

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp (hunk 1 above)
Comment (v4hn) [reply moveit_ros/planning/plan_execution/src/plan_execution.cpp]: That's a standard signalling issue: Either you respond to signals that might have been sent even before the call started (my proposal to remove the line), or there is a negligible time between calling the method and `stop()` actually triggering a preemption. There's ways around it of course, but I don't see the need for that here at the moment.

I actually prefer to reset the signal, but `planAndExecute` has the same problem with my old patch.
So now I basically added your previous patch to fix that case too.

**Comment by v4hn** [comment]
> Please feel free to discard particularly the last one if you don't like the proposed single-line format.

I originally wrote it that way too, but split the statements for ease of readability.

Thanks for the immediate review. :-)
I'll merge once CI succeeds.

**Pull request merged**